spring-webflux/src/main/java/org/springframework/web/reactive/function/BodyInserters.java

```diff
@@ -373,12 +373,12 @@ else if (adapter != null) {
 			publisher = Mono.just(body);
 		}
 		MediaType mediaType = outputMessage.getHeaders().getContentType();
-		return context.messageWriters().stream()
-				.filter(messageWriter -> messageWriter.canWrite(bodyType, mediaType))
-				.findFirst()
-				.map(BodyInserters::cast)
-				.map(writer -> write(publisher, bodyType, mediaType, outputMessage, context, writer))
-				.orElseGet(() -> Mono.error(unsupportedError(bodyType, context, mediaType)));
+		for (HttpMessageWriter<?> messageWriter : context.messageWriters()) {
+			if (messageWriter.canWrite(bodyType, mediaType)) {
+				return write(publisher, bodyType, mediaType, outputMessage, context, cast(messageWriter));
```

Comment:
compared to the `BodyExtractor` case, I find that `cast` call is a little too buried among all these parameters, making it less visible. would you mind extracting it to an intermediate local variable? this should be efficiently optimized away by the JIT compiler, so that is purely from a readability perspective.